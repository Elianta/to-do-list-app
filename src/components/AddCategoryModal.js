import React from 'react';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import Modal from 'react-modal';
import {closeModals} from '../actions/modals';
import {addNestedCategory} from '../actions/categories';

export class AddCategoryModal extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            category: ''
        };
        this.handleSaveNewCategory = this.handleSaveNewCategory.bind(this);
    }

    handleSaveNewCategory(e) {
        e.preventDefault();
        const categoryName = this.state.category;
        this.props.addNestedCategory({name: categoryName, parentID: this.props.modals.editableCategory.id});
        this.props.closeModals();
    }

    render() {
        return (
            <Modal
                isOpen={this.props.modals.isAddCategoryModalOpen}
                onRequestClose={this.props.handleModalClose}
                contentLabel="Add Category"
                closeTimeoutMS={200}
                className="modal"
            >
                <h3 className="modal__title">{`Add nested category into '${this.props.modals.editableCategory.name}'`}</h3>
                <form data-testid="form" className="modal__form" onSubmit={this.handleSaveNewCategory}>
                    <input
                        type="text"
                        name="category"
                        autoFocus={true}
                        className="modal__input"
                        data-testid="name"
                        onChange={(e) => {
                            this.setState({category: e.currentTarget.value.trim()});
                        }}
                    />
                    <button className="button  modal__btn">Save</button>
                </form>
            </Modal>
        );
    }
}

const mapStateToProps = (state) => ({
    modals: state.modals
});

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
        addNestedCategory,
        closeModals
    }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(AddCategoryModal);
