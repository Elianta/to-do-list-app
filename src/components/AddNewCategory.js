import React from 'react';
import {connect} from 'react-redux';
import {addCategory} from '../actions/categories';

class AddNewCategory extends React.Component {
    constructor(props) {
        super(props);
        this.handleAddNewCategory = this.handleAddNewCategory.bind(this);
        this.state = {
            error: undefined
        };
    }

    handleAddNewCategory(e) {
        e.preventDefault();
        const categoryName = e.target.elements.category.value.trim();
        if (!categoryName) {
            this.setState(() => ({error: 'Enter valid value to add item'}));
        } else {
            this.setState(() => ({
                error: ''
            }));
            this.props.dispatch(addCategory(categoryName));
            e.target.elements.category.value = '';
        }
    }

    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form className="add-item-form" onSubmit={this.handleAddNewCategory}>
                    <input
                        className="add-item-form__input"
                        type="text"
                        name="category"
                        placeholder="Enter category title..."
                    />
                    <button className="button">Add</button>
                </form>
            </div>
        );
    }
}

export default connect()(AddNewCategory);
