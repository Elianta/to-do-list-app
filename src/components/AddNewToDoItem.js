import React from "react";

export default class AddNewToDoItem extends React.Component {
  constructor(props) {
    super(props);
    this.handleAddNewToDoItem = this.handleAddNewToDoItem.bind(this);

    this.state = {
      error: undefined
    };
  }

  handleAddNewToDoItem(e) {
    e.preventDefault();
    const toDoItem = e.target.elements.toDoItem.value.trim();
    const error = this.props.handleAddNewToDoItem(this.props.categoryID, toDoItem);

    this.setState(() => ({ error }));

    if (!error) {
      e.target.elements.toDoItem.value = '';
    }
  }

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form className="add-item-form" onSubmit={this.handleAddNewToDoItem}>
          <input className="add-item-form__input" type="text" name="toDoItem" placeholder="Enter item title"/>
          <button className="button">Add</button>
        </form>
      </div>
    )
  }
}
