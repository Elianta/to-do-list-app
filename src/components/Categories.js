import React from 'react';
import Category from './Category.js';

class Categories extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            categoryToMoveTaskInId: null
        };
        this.handleSpecifyCategoryToMove = this.handleSpecifyCategoryToMove.bind(this);
    }

    handleSpecifyCategoryToMove(categoryID) {
        this.setState(() => ({
            categoryToMoveTaskInId: categoryID
        }));
        this.props.handleSpecifyCategoryToMove(categoryID);
    }

    render() {
        let categories = this.props.categories.map((category) => {
            return (
                <Category
                    node={category}
                    children={category.children}
                    handleAddNestedCategory={this.props.handleAddNestedCategory}
                    key={category.name}
                    handleEditCategory={this.props.handleEditCategory}
                    handleDeleteCategory={this.props.handleDeleteCategory}
                    mode={this.props.mode}
                    updateProgressBar={this.props.updateProgressBar}
                    categoryID={this.props.categoryID}
                    handleSpecifyCategoryToMove={this.handleSpecifyCategoryToMove}
                    categoryToMoveTaskInId={this.state.categoryToMoveTaskInId}
                    transferTaskIn={this.state.categoryToMoveTaskInId === category.id}
                />
            );
        });

        return (
            <div className="categories__container">
                <ul className="categories__list">
                    {categories}
                </ul>
            </div>
        );
    }
}

export default Categories;
