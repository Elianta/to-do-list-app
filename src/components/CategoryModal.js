import React from 'react';
import Modal from 'react-modal';

const CategoryModal = (props) => (
    <Modal
        isOpen={props.categoryModalDisplayed}
        onRequestClose={props.handleModalClose}
        contentLabel="Selected Category"
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">{props.title}</h3>
        <form className="modal__form" onSubmit={props.handleSaveCategory}>
            <input
                type="text"
                name="category"
                autoFocus={true}
                className="modal__input"
                defaultValue={props.modalMode === 'edit' ? props.categoryName : ''}
            />
            <button className="button  modal__btn">Save</button>
        </form>
    </Modal>
);

export default CategoryModal;
