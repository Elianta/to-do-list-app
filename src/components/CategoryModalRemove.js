import React from 'react';
import Modal from 'react-modal';

const CategoryModalRemove = (props) => (
    <Modal
        isOpen={props.categoryModalRemoveDisplayed}
        onRequestClose={props.handleModalClose}
        contentLabel="Selected Category"
        closeTimeoutMS={200}
        className="modal"
    >
        <h3 className="modal__title">{`Do you want to remove '${props.categoryName}'?`}</h3>
        <p className="modal__content">(All nested categories will be also removed)</p>
        <button className="button  modal__btn" onClick={props.handleRemoveConfirm}>OK</button>
        <button className="button  modal__btn" onClick={props.handleModalClose}>Cancel</button>
    </Modal>
);

export default CategoryModalRemove;
