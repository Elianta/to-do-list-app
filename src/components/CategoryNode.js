export default class CategoryNode {
  constructor(name, id) {
    this.name = name;
    this.id = id;
    this.children = [];
    this.tasks = [];
  }
}