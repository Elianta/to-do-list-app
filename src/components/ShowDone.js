import React from 'react';
import {connect} from 'react-redux';
import {setShowDoneFilter} from '../actions/filters';
import {bindActionCreators} from 'redux';

export class ShowDone extends React.Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(e) {
        const showDone = e.target.checked;
        this.props.setShowDoneFilter(showDone);
        this.changeBrowserHistory(showDone);
    }

    changeBrowserHistory(showDone) {
        if (this.props.filters.text) {
            this.props.history.replace({
                pathname: this.props.location.pathname,
                search: `?showdone=${showDone}&text=${this.props.filters.text}`,
            });
        } else {
            this.props.history.replace({
                pathname: this.props.location.pathname,
                search: `?showdone=${showDone}`,
            });
        }
    }

    render() {
        return (
            <div className="option">
                <input
                    data-testid="show-done"
                    type="checkbox"
                    className="option__input"
                    id="show-done"
                    checked={this.props.filters.showDone}
                    onChange={this.handleChange}
                />
                <label className="option__label" htmlFor="show-done">Show done</label>
            </div>
        );
    }
}

const mapStateToProps = (state) => ({
    filters: state.filters
});

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
        setShowDoneFilter
    }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(ShowDone);
