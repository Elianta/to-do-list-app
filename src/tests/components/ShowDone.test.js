import React from 'react';
import {shallow} from 'enzyme';
import {ShowDone} from '../../components/ShowDone';

test('should render ShowDone correctly', () => {
    const filters = {
        showDone: true
    };
    const wrapper = shallow(<ShowDone filters={filters}/>);
    expect(wrapper).toMatchSnapshot();
});

test('should send correct showdone value on change', () => {
    const setShowDoneFilter = jest.fn();
    const filters = {
        showDone: true
    };
    const wrapper = shallow(<ShowDone filters={filters} setShowDoneFilter={setShowDoneFilter}/>);
    const spy = jest.spyOn(wrapper.instance(), 'changeBrowserHistory').mockImplementation(() => {});
    const showdone = wrapper.find('[data-testid="show-done"]');

    showdone.simulate('change', {target: {checked: !filters.showDone}});

    expect(spy).toHaveBeenCalledWith(!filters.showDone);
    expect(wrapper.instance().props.setShowDoneFilter).toHaveBeenCalledWith(!filters.showDone);
});
