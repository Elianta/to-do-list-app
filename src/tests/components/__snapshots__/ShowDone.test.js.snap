// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ShowDone correctly 1`] = `
<div
  className="option"
>
  <input
    checked={true}
    className="option__input"
    data-testid="show-done"
    id="show-done"
    onChange={[Function]}
    type="checkbox"
  />
  <label
    className="option__label"
    htmlFor="show-done"
  >
    Show done
  </label>
</div>
`;
